src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,46 @@
 import React from 'react';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
+import { NoComments } from '../NoComments';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+import { Post } from '../../types/Post';
+
+type Props = {
+  post: Post,
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    userId,
+    title,
+    body,
+    user,
+    comments,
+  } = post;
+
+  return (
+    <div className="PostInfo">
+      <div className="PostInfo__header">
+        <h3 className="PostInfo__title">
+          {title}
+        </h3>
+
+        <p>
+          {' Posted by  '}
+
+          {user && <UserInfo user={user} key={userId} />}
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      {
+        (comments.length > 0 && <CommentList comments={comments} />)
+        || <NoComments />
```

Comment:
this one is hard to read. I catch idea, how U use logical operators
maybe just conditional will be more readable
comments.length > 0
? `<CommentList comments={comments} />`
: `<NoComments />`